pkg/controller/common.go

```diff
@@ -98,6 +103,28 @@ type azReconciler interface {
 	getAzClient() azClientSet.Interface
 }
 
+type retryInfo struct {
```

Comment:
The motivation behind my [recommendation of the `retryInfo` struct](https://github.com/abhisheksinghbaghel/azuredisk-csi-driver/pull/158#discussion_r703922769) was to reduce usage back down to one map, not to encapslate the two. In other words, create a `sync.Map` of `retryInfo` structs. This way we avoid the double lookup each time we need to recalculate the next requeue time.

I do like the encapsulation of the maps you've done here though. It does make for a cleaner interface for the caller; that it, is doesn't have to do the messy lookup and cast. Perhaps, something more like this:

```go
type retryInfoEntry {
    mutex   sync.Mutex
    backoff retry.Backoff
}

type retryInfo {
    entries sync.Map
}

...

func (r *retryInfo) nextRequeue(objectName string) time.Duration {
    v, _ := r.entries.LoadOrStore(objectName, &retryInfoEntry{})
    entry := v.(*retryInfoEntry)
    entry.Lock()
    defer entry.Unlock()
    return entry.backoff.Step()
}
```

Reply:
Ah, that makes a lot of sense. Will do :)